test/test_buildapi_client.py

```diff
@@ -0,0 +1,158 @@
+            auth=None)
+
+    @patch('requests.post', return_value=mock_response(POST_RESPONSE, 200))
+    def test_call_with_dry_run(self, post):
+        """make_retrigger_request should return None when dry_run is True."""
+        self.assertEquals(
+            buildapi_client.make_retrigger_request("repo", "1234567", auth=None, dry_run=True), None)
+        # make_retrigger_request should not call requests.post when dry_run is True
+        assert post.call_count == 0
+
+    @patch('requests.post', return_value=mock_response(POST_RESPONSE, 200))
+    def test_call_with_different_priority(self, post):
+        """make_retrigger_request should call requests.post with the right priority."""
+        buildapi_client.make_retrigger_request("repo", "1234567", priority=2, auth=None, dry_run=False)
+        post.assert_called_once_with(
+            '%s/%s/request' % (HOST_ROOT, "repo"),
+            headers={'Accept': 'application/json'},
+            data={'count': 1, 'priority': 2, 'request_id': '1234567'},
+            auth=None)
+
+    @patch('requests.post', return_value=mock_response(POST_RESPONSE, 200))
+    def test_call_with_different_count(self, post):
+        """make_retrigger_request should call requests.post with the right count."""
+        buildapi_client.make_retrigger_request("repo", "1234567", count=10, auth=None, dry_run=False)
+        post.assert_called_once_with(
+            '%s/%s/request' % (HOST_ROOT, "repo"),
+            headers={'Accept': 'application/json'},
+            data={'count': 10, 'priority': 0, 'request_id': '1234567'},
+            auth=None)
+
+
+class TestMakeCancelRequest(unittest.TestCase):
+
+    """Test that make_cancel_request makes the right DELETE requests."""
+
+    @patch('requests.delete', return_value=Mock())
+    def test_call_without_dry_run(self, delete):
+        """make_cancel_request should call requests.delete."""
+        buildapi_client.make_cancel_request("repo", "1234567", auth=None, dry_run=False)
+
+        # We expect that make_cancel_request will call requests.delete
+        # once with the following arguments
+        delete.assert_called_once_with(
+            '%s/%s/request/%s' % (HOST_ROOT, "repo", "1234567"),
+            auth=None)
+
+    @patch('requests.delete', return_value=Mock())
+    def test_call_with_dry_run(self, delete):
+        """make_cancel_request should return None when dry_run is True."""
+        self.assertEquals(
+            buildapi_client.make_cancel_request("repo", "1234567", auth=None, dry_run=True), None)
+        # make_cancel_request should not call requests.delete when dry_run is True
+        assert delete.call_count == 0
+
+class TestMakeQueryRepositoriesRequest(unittest.TestCase):
+
+    """Test that make_query_repositories_request makes the right GET requests."""
+
+    @patch('requests.get', return_value=Mock())
+    def test_call_without_dry_run(self, get):
+        """trigger_arbitrary_job should call requests.post."""
```

Comment:
You forgot to change the docstring in your tests :-) (I always forget the docstrings)